java/test/jglome/GlomeTest.java

```diff
@@ -0,0 +1,153 @@
+import java.security.interfaces.XECPublicKey;
+import java.util.Arrays;
+import java.util.Optional;
+import jglome.Glome.WrongTagException;
+import org.junit.jupiter.api.Test;
+import jglome.Glome.CounterOutOfBoundsException;
+import jglome.Glome.MinPeerTagLengthOutOfBoundsException;
+import jglome.Glome.StaticSecureRandom;
+
+/**
+ * Class, which contains tests for main.jglome.Glome functionality.
+ */
+public class GlomeTest {
+
+  final static int N_TEST_VECTORS = 2;
+
+  TestVector[] testVectors = {testVector1, testVector2};
+  Glome[][] glomeManagers = new Glome[N_TEST_VECTORS][2]; // first is for A, second is for B
+  KeyPair[] aKeys = new KeyPair[N_TEST_VECTORS];
+  KeyPair[] bKeys = new KeyPair[N_TEST_VECTORS];
+
+  GlomeTest() throws MinPeerTagLengthOutOfBoundsException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      aKeys[i] = deriveKeysFromPrivate(testVectors[i].kah);
+      bKeys[i] = deriveKeysFromPrivate(testVectors[i].kbh);
+      glomeManagers[i][0] = new Glome((XECPublicKey) bKeys[i].getPublic(),
+          (XECPrivateKey) aKeys[i].getPrivate(), 32);
+      glomeManagers[i][1] = new Glome((XECPublicKey) aKeys[i].getPublic(),
+          (XECPrivateKey) bKeys[i].getPrivate(), 28);
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenMinPeerTagLengthIsOutOfBounds() {
+    try {
+      new Glome((XECPublicKey) aKeys[0].getPublic(), (XECPrivateKey) bKeys[0].getPrivate(), 0);
+    } catch (MinPeerTagLengthOutOfBoundsException e) {
+      assertEquals(e.getMessage(), "minPeerTagLength argument should be in [1..32] range. Got 0.");
+    }
+
+    try {
+      new Glome((XECPublicKey) aKeys[0].getPublic(), (XECPrivateKey) bKeys[0].getPrivate(), 33);
+    } catch (MinPeerTagLengthOutOfBoundsException e) {
+      assertEquals(e.getMessage(), "minPeerTagLength argument should be in [1..32] range. Got 33.");
+    }
+  }
+
+  @Test
+  public void testShouldFailWhenCounterIsOutOfBounds() {
+    TestVector vector = testVectors[0];
+    try {
+      assertArrayEquals(vector.tag, glomeManagers[0][0].generateTag(vector.msg, -1));
+    } catch (CounterOutOfBoundsException e) {
+      assertEquals(e.getMessage(), "Counter should be in [0..255] range. Got -1.");
+    }
+
+    try {
+      assertArrayEquals(vector.msg, glomeManagers[0][0].generateTag(vector.msg, 256));
+    } catch (CounterOutOfBoundsException e) {
+      assertEquals(e.getMessage(), "Counter should be in [0..255] range. Got 256.");
+    }
+  }
+
+  @Test
+  public void derivedKeyShouldEqualOriginalKey() {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      assertEquals(aKeys[i].getPublic(), glomeManagers[i][0].userKeys().getPublic());
+      assertEquals(bKeys[i].getPublic(), glomeManagers[i][1].userKeys().getPublic());
+    }
+  }
+
+  @Test
+  public void testTagGeneration() throws CounterOutOfBoundsException {
+    for (int i = 0; i < N_TEST_VECTORS; i++) {
+      TestVector vector = testVectors[i];
+      int sender = i % 2 == 0 ? 0 : 1;
```

Comment:
Why `i % 2 == 0 ? 0 : 1;` instead of `i % 2`. Also consider what would happen if a third test vector was to be added.